src/System.IO.FileSystem/src/System/IO/Win32FileSystemObject.cs

```diff
@@ -64,7 +64,7 @@ public DateTimeOffset CreationTime
                 {
                     EnsureDataInitialized();
                     long dt = ((long)(_data.ftCreationTimeHigh) << 32) | ((long)_data.ftCreationTimeLow);
-                    return DateTimeOffset.FromFileTime(dt);
+                    return DateTimeOffset.FromFileTime(dt).ToLocalTime();
```

Comment:
Slight addition to my above comment:
The windows calls to ToLocalTime() may be removed, but <b>only</b> if the DateTimeOffsets returned use .LocalTime instead of .DateTime to convert to a DateTime object. Otherwise we get errors.

I apologize for the potentially confusing comments; there are so many places where the conversion can go wrong it's difficult to nail down exactly which parts should be fixed. I'll add some more specific tests.